packages/hardhat-core/test/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -1511,13 +1517,18 @@ describe("Hardhat module", function () {
         }
       });
 
+      const isEdr = process.env.HARDHAT_EXPERIMENTAL_VM_MODE === "edr";
+
       describe("hardhat_setBalance", function () {
         it("should reject an invalid address", async function () {
           await assertInvalidArgumentsError(
             this.provider,
             "hardhat_setBalance",
             ["0x1234", "0x0"],
-            'Errors encountered in param 0: Invalid value "0x1234" supplied to : ADDRESS'
+            // TODO: https://github.com/NomicFoundation/edr/issues/104
+            `${
+              isEdr ? "" : "Errors encountered in param 0: "
+            }Invalid value "0x1234" supplied to : ADDRESS`
```

Comment:
Is there a blocker to make these work without special casing EDR?

Reply:
Ah found the answer in Slack:

> EDR is currently not capable of mentioning which parameter index failed to parse. This would require us to handroll deserialization for the enum variants

Reply:
The issue already exists in the form of [#104](https://github.com/NomicFoundation/edr/issues/104)